### DISCUSSION TITLE
Debugging the SWA in VS Code

### DISCUSSION THREAD

**Discussion by crashAlgorithm** [Azure/static-web-apps#730] [Q&A] [answered]
Hello,

I'm wondering if the Azure Static Web Apps extension for VS Code is supposed to give me the ability to debug both the front-end code and Azure functions in my project at the same time.

When I start debugging with the auto-generated SWA debugging profile, the functions host has to already be started for the SWA emulator to start correctly. Doing so this way does not allow debugging of the functions - only the Angular code I am working with.

If I manually ng serve and then debug with the "Attach to .NET functions" profile, I can start the SWA emulator manually, but I'm only able to hit breakpoints in the Azure functions, not in the Angular code.

**Comment by alexweininger** [comment, marked as answer]
Hi, thank you for your feedback and bringing this to our attention!

I've created an issue: https://github.com/microsoft/vscode-azurestaticwebapps/issues/617 on the Azure Static Web Apps extension for VS Code repository. We'll track it there.

I just tested and verified a workaround that might work until a fix is released. In your `launch.json`, change `Attach to .NET Functions` to `Attach to Node Functions`. We need to change a hardcoded string to accommodate for other languages, but for now that should work.

**Reply by crashAlgorithm** [reply]
Thanks for the quick response! Your proposed workaround did fix the issue for me. 

I'm loving Static Web Apps and the VSCode tools, keep up the great work.

**Reply by alexweininger** [reply]
> Thanks for the quick response! Your proposed workaround did fix the issue for me.

No problem, glad the temporary fix worked!

> I'm loving Static Web Apps and the VSCode tools, keep up the great work.

Yay! Thank you very much for your kind words! ❤️
